customer-samples/RabbitMQ.Sample/Receive/Receive.cs

```diff
@@ -0,0 +1,95 @@
+                                         autoDelete: false,
+                                         arguments: null);
+
+                    var consumer = new EventingBasicConsumer(channel);
+                    consumer.Received += (model, ea) =>
+                    {
+                        var body = ea.Body.ToArray();
+                        var message = Encoding.UTF8.GetString(body);
+
+                        // Read the basic property headers and extract the Datadog properties
+                        var headers = ea.BasicProperties.Headers;
+                        ulong? parentSpanId = null;
+                        ulong? traceId = null;
+                        SamplingPriority? samplingPriority = null;
+
+                        // Parse parentId
+                        if (headers?[HttpHeaderNames.ParentId] is byte[] parentSpanIdBytes)
+                        {
+                            var parentSpanIdString = Encoding.UTF8.GetString(parentSpanIdBytes);
+                            if (ulong.TryParse(parentSpanIdString, System.Globalization.NumberStyles.Integer, CultureInfo.InvariantCulture, out var result))
+                            {
+                                parentSpanId = result;
+                            }
+                        }
+
+                        // Parse traceId
+                        if (headers?[HttpHeaderNames.TraceId] is byte[] traceIdBytes)
+                        {
+                            var traceIdString = Encoding.UTF8.GetString(traceIdBytes);
+                            if (ulong.TryParse(traceIdString, System.Globalization.NumberStyles.Integer, CultureInfo.InvariantCulture, out var result))
+                            {
+                                traceId = result;
+                            }
+                        }
+
+                        // Parse samplingPriority
+                        if (headers?[HttpHeaderNames.SamplingPriority] is byte[] samplingPriorityBytes)
+                        {
+                            var samplingPriorityString = Encoding.UTF8.GetString(samplingPriorityBytes);
+                            if (Enum.TryParse<SamplingPriority>(samplingPriorityString, out var result))
+                            {
+                                samplingPriority = result;
+                            }
+                        }
+
+                        // Create a new SpanContext to represent the distributed tracing information
+                        SpanContext propagatedContext = null;
+                        if (parentSpanId.HasValue && traceId.HasValue)
+                        {
+                            propagatedContext = new SpanContext(traceId, parentSpanId.Value, samplingPriority);
+                        }
+
+                        // Start a new Datadog span
+                        using (var scope = Tracer.Instance.StartActive("rabbitmq-consume", propagatedContext))
```

Comment:
Consider adding something like `Thread.Sleep(...)` to do fake work. Otherwise the spans will be tiny in the APM UI. Maybe also show how to set a few tags like the queue or topic names.

Reply:
Thanks, done in 658137438ce43129f45c24aef602c6c2f7700291